src/Waypoint/WaypointList.cpp

```diff
@@ -62,3 +62,30 @@ WaypointList::SortByDistance(const GeoPoint &location) noexcept
 {
   std::sort(begin(), end(), WaypointDistanceCompare(location));
 }
+
+class WaypointNameCompare
```

Comment:
What's the point of this class? It has no fields; this should better be a standalone function with no `this` pointer overhead, or maybe even a lambda.

Reply:
I have changed it to a lambda